### PR TITLE
design-docs: timetable: unify arrival times and margin types

### PR DESCRIPTION
**Pull request by multun**

### CODE REVIEW

File: content/docs/reference/design-docs/timetable.en.md [hunk 1]
```diff
@@ -86,21 +127,44 @@ path:
...
-engineering_margins:
- - {from: b, to: c, value: "2%", distribution: MARECO}
+stops:
+ - {at: a, for: PT5M}
 
-standard_margins:
-  # the begin and end waypoints are always implicitly added
+# there are two mutually exclusive ways to add time constraints:
+# margins and arrival times. Both methods share parameters, such
+# as the distribution, intermediate waypoints, and transitions.
```
Comment (flomonster) [inline content/docs/reference/design-docs/timetable.en.md:135]: We should check that these assertion are correct:
- Having a single way of computing margins
- Having two mutually exclusive ways to add time constraints

File: content/docs/reference/design-docs/timetable.en.md [hunk 2]
```diff
@@ -86,21 +127,44 @@ path:
...
+  # time constraints apply on the entire path. intermediate
+  # waypoints delimit zones which can be configured with distinct
+  # time constraints.
+  intermediate_waypoints: [b, c]
+
+  # can be BEFORE, AFTER or AUTO. If AUTO, the transition type is
+  # unspecified, and left for the system to figure out.
+  # There has to be as many transitions as intermediate points.
+  # If the field is omited, the default transition type is AUTO.
+  transition_types: ["AFTER", "AUTO"]
```
Comment (flomonster) [inline content/docs/reference/design-docs/timetable.en.md:149]: It feels like it will never be specified by the user, and so never used.

**Comment by flomonster** [comment]
Idea of new model:

- We could have only one type of margin (like the engineering margin). 
- Each margin can have an associated algorithm (linear or mareco).
- This list of margins can be reduced to two sets of margins (linear and mareco)
- This set of margins can be applied using the "AUTO" method avoiding problems of speed continuity.

File: content/docs/reference/design-docs/timetable.en.md (hunk 2 above)
Comment (multun) [reply content/docs/reference/design-docs/timetable.en.md:149]: It's the only difference between construction margins and baseline margins

**Comment by flomonster** [comment]
We asked the users questions and this modified our basic proposal. https://github.com/osrd-project/osrd-website/pull/162/commits/a0ea66286ceff8012f9e8946695bcd3292daa048

**Comment by multun** [comment]
I'd love to further simplify the proposal given the user feedback. I can do this next week, or someone can give it a try until then

File: content/docs/reference/design-docs/timetable.en.md (hunk 2 above)
Comment (multun) [reply content/docs/reference/design-docs/timetable.en.md]: this is gone, as there are no more construction margins

File: content/docs/reference/design-docs/timetable/index.en.md [hunk 3]
```diff
@@ -129,19 +135,28 @@ path:
...
+  # algorithm used to distribute time over space
+  distribution: MARECO
+
+  # time constraints apply on the entire path. boundaries delimit
+  # zones which can be configured with distinct time constraints.
+  boundaries: [b, c]
+
+  # there are as many constraints as zones: if there are N boundaries,
+  # there are be N + 1 zones, and thus there must be N + 1 constraints
+  contraints:
```
Comment (Khoyo) [inline content/docs/reference/design-docs/timetable/index.en.md:153]:
```suggestion
  constrains:
```

**Pull request merged**